spec/features/sign_up_spec.rb

```diff
@@ -0,0 +1,72 @@
+feature 'FEATURE: sign up' do
+
+  scenario 'Can click to sign up from home page' do
+    visit('/')
+    click_link('sign_up')
+    expect(page.current_path).to eq "/users/new"
+  end
+
+  let(:name) { 'J T Kirk' }
+  let(:email) { 'JTK@test.com' }
+  let(:user_name) { 'TheShat' }
+  let(:password) { 'password' }
+  let(:password_confirmation) { 'password' }
+
+  context 'Already signed in' do
+
+    before do
+      User.create(name: name, email: email, user_name: user_name, password: password, password_confirmation: password_confirmation)
+    end
+
+    scenario "cannot click to sign up" do
+      sign_in(email, password)
+      expect(page).not_to have_selector("#sign_up")
+    end
+  end
+
+  scenario 'Can enter user details and create new user' do
+    sign_up('James T Kirk', 'JTK@test.com', 'TheShat', 'password', 'password')
```

Comment:
Might be an opportunity for keyword arguments to make it clearer what all these things are.